contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,144 @@
+
+## What's so difficult about mobile session replay
+
+Others have complained about [the lack of good mobile replay options](https://medium.com/goodones/15-years-later-there-is-still-no-good-session-replay-for-ios-f8d335999737). Why is that the case?
+
+### 1. Multiple platforms
+
+The secret about web session replay industry-wide is that it largely relies on a single open source library to work: [rrweb](https://github.com/rrweb-io/rrweb). It includes tools for recording web interactions and state changes, structuring session data, and playback. 
+
+Unfortunately, rrweb for mobile doesn't exist. To build mobile session replay, we needed to do all the work ourselves, and when compared to the web, this is a lot of work. This is because, instead of a single JavaScript library, language, and SDK, mobile requires multiple (like iOS, Android, React Native, and Flutter). 
+
+There are even breaking differences within platforms. For example, Jetpack Compose uses a compositional model for UI, different from Android's traditional view-based model. This means you need to develop separate ways of doing replays when using it. iOS has a similar problem with SwiftUI.
+
+### 2. Performance
+
+Desktops are a lot more powerful than phones. Because of this, we need to be much more sensitive about performance. 
+
+If you've ever tried to record your phone screen, you would know its impact on performance. Apps take longer to load, animations become choppy, reactivity degrades, and your phone heats up.
+
+Any that degrades the experience this much is not an option for many developers.
+
+### 3. Privacy
+
+A big difference between web and mobile apps is the DOM. 
+
+In web apps, this provides a hierarchical tree structure that represents and uniquely identifies elements. The web also has standardized elements, like `<input type="password">`. Combining these makes it easier to identify, mask, and exclude sensitive elements on the web.
+
+Mobile doesn't have standardized structures or elements. Accessibility identifiers are also inconsistently implemented. This means identifying, masking, and excluding content is a lot more tricky.
+
+### 4. Testing
+
+We're big fans of [dogfooding](/product-engineers/dogfooding) at PostHog. Often, we are our own best users. As you might know, we don't have a mobile app. 
+
+During development, we would need to rely on demo and open source apps. We risk creating something that doesn't work well for larger, production-quality apps.
+
+Developing a high-quality mobile replay product means relying more on our users and their feedback.
+
+## The prerequisites for mobile session replay
+
+First, none of this would be possible without [Manoel](/community/profiles/30206). We had mobile experience, but not the dedicated mobile SDK experience needed for this big of a feature. Manoel had that experience and this was only possible thanks to him.
```

Comment:
The second bit of this paragraph is not super clear tbh. I think we can just remove
```suggestion
First, none of this would be possible without [Manoel](/community/profiles/30206). 
```